src/components/Chatbot/Chatbot.tsx

```diff
@@ -23,90 +25,180 @@ const NewConversationButton = styled(PrimaryButton)(() => ({
 }));
 
 export const Chatbot = () => {
-  const [conversation, setConversation] = useState<Message[]>([]);
+  const [messages, setMessages] = useState<MessageBubbleProps[]>([]);
   const [error, setError] = useState(false);
   const [loading, setLoading] = useState(false);
+  const [responding, setResponding] = useState(false);
+  const [conversationCreated, setConversationCreated] = useState<string | null>(
+    null
+  );
+  const [errorMessage, setErrorMessage] = useState('Unknown error');
 
-  const addMessageToConversation = async (message: Message) => {
-    setConversation([...conversation, message]);
+  const loadConversationFromBackend = async () => {
+    setLoading(true);
     try {
       setLoading(true);
-      setConversation([
-        ...conversation,
-        message,
-        { originBot: true, text: '...' }
-      ]);
-      await messageChatbot(message.text).then((response) => {
+      loadConversation().then((response) => {
```

Comment:
This is the reason why the loading indicator does not work.
As I pointed out in another comment, you are mixing `async-await` and Promises (`then()`, etc.).
The outer `finally` sets loading to false before this promise resolves.

Reply:
aaa oki